### PR TITLE
Fix to solve the nil pointer panic.

### PR DESCRIPTION
**Pull request by rmrichesjr1**

### CODE REVIEW

**Comment by gemcdaniel** [comment]
@rmrichesjr1 can you update the version of Go in the `go.mod` to `1.21.12`. In go 1.21 and newer, this should follow the syntax 1.N.P. Thanks!

**Review by gemcdaniel — APPROVED** [review]

**Review by developingjames — DISMISSED** [review]

File: internal/provider/resource_apply_module.go [hunk 1]
```diff
@@ -300,16 +300,19 @@ func (t *applyModuleResource) Read(ctx context.Context,
...
-		state.ModuleVersion = types.StringValue(*currentApplied.moduleVersion)
-	} else {
-		state.ModuleVersion = types.StringNull()
+	// If there is no current state version, currentApplied can be nil.
+	if currentApplied != nil {
+		// Update the state with the computed module source and version.
+		if currentApplied.moduleSource != nil {
+			state.ModuleSource = types.StringValue(*currentApplied.moduleSource)
+		} else {
+			state.ModuleSource = types.StringNull()
```
Comment (bbdev442) [inline internal/provider/resource_apply_module.go:309]: Shouldn't the module source and module version be set to a null string value when the current applied is null?

Comment (rmrichesjr1) [reply internal/provider/resource_apply_module.go]: Implemented what I understood was the intent of this review item.

**Comment by rmrichesjr1** [comment]
Successfully tested commit 7134567 on all three sequences that had shown the nil pointer bug.

**Review by amanshallow — DISMISSED** [review]

File: internal/provider/resource_apply_module.go [hunk 2]
```diff
@@ -300,13 +300,14 @@ func (t *applyModuleResource) Read(ctx context.Context,
...
-	if currentApplied.moduleSource != nil {
+	// If there is no current state version, currentApplied can be nil.
+	// If available/possible, update the state with the computed module source and version.
+	if (currentApplied != nil) && (currentApplied.moduleSource) != nil {
 		state.ModuleSource = types.StringValue(*currentApplied.moduleSource)
 	} else {
 		state.ModuleSource = types.StringNull()
 	}
-	if currentApplied.moduleVersion != nil {
+	if (currentApplied != nil) && (currentApplied.moduleVersion) != nil {
```
Comment (bbdev442) [inline internal/provider/resource_apply_module.go:310]: Parenthesis are not needed around `(currentApplied.moduleSource)` (same for below)

**Review by bbdev442 — APPROVED** [review]

**Pull request merged**